src/connectors/NativeConnector.sol

```diff
@@ -1,27 +1,24 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.17;
 
-import {InsufficientFunds, RefundFailed, ZeroValueCall} from "../errors/ConnectorErrors.sol";
-import {SharedStructs} from "../lib/SharedStructs.sol";
-import {UpgradeableBase} from "./UpgradeableBase.sol";
-import {Receiver} from "./Receiver.sol";
+import {ConnectorUpgradeableBase} from "./ConnectorUpgradeableBase.sol";
+import {Receiver} from "../lib/Receiver.sol";
 import {NativeConnectorLogic} from "../connectors/NativeConnectorLogic.sol";
 import {Constants} from "../lib/Constants.sol";
 import {TokenState} from "./TokenState.sol";
 import {BridgeBase} from "../lib/BridgeBase.sol";
 
-contract NativeConnector is UpgradeableBase, Receiver, NativeConnectorLogic {
+contract NativeConnector is ConnectorUpgradeableBase, Receiver, NativeConnectorLogic {
     function initialize(BridgeBase _bridge, address token_on_other_network) public initializer {
         require(
             address(_bridge) != address(0) && address(Constants.NATIVE_TOKEN_ADDRESS) != address(0)
                 && token_on_other_network != address(0),
             "TokenConnectorBase: invalid bridge, token, or token_on_other_network"
         );
-        __BridgeConnectorBase_init(address(_bridge));
+        __ConnectorUpgradeableBase_init(address(_bridge));
```

Comment:
The only problem I have with this is the following: 
 - When we finalize an epoch, we check whether the connector is able to pay. This is the only criterion to be eligible.
  - If connectors pass their ownership to the bridge on deployment they can't be upgraded by their deployers anymore.
  - In addition, we don't have a way, as admins, to delete(delist) connectors if we or someone else fucks up something. 
  - This in turn kind of locks a token to not being bridgeable anymore.